Rebus.RabbitMq/RabbitMq/RabbitMqTransport.cs

```diff
@@ -447,49 +473,132 @@ CustomQueueingConsumer InitializeConsumer()
+            if (headers.TryGetValue(RabbitMqHeaders.DeliveryMode, out var deliveryModeVal))
+            {
+                if (byte.TryParse(deliveryModeVal, out var deliveryMode) && deliveryMode > 0 && deliveryMode <= 2)
                 {
-                    EnsureQueueIsInitialized(destinationAddress, model);
+                    props.DeliveryMode = deliveryMode;
                 }
+            }
 
-                model.BasicPublish(exchange, routingKey.RoutingKey, props, message.Body);
+            if (headers.TryGetValue(RabbitMqHeaders.Expiration, out var expiration))
+            {
+                if (TimeSpan.TryParse(expiration, out var timeToBeDelivered))
+                { 
+                    props.Expiration = timeToBeDelivered.TotalMilliseconds.ToString("0");
+                }
             }
+
+            if (headers.TryGetValue(RabbitMqHeaders.Priority, out var priorityVal))
+            {
+                if (byte.TryParse(priorityVal, out var priority))
+                {
+                    props.Priority = priority;
+                }
+            }
+
+            if (headers.TryGetValue(RabbitMqHeaders.Timestamp, out var timestampVal))
+            {
+                if (DateTimeOffset.TryParse(timestampVal, out var timestamp))
+                {
+                    // Unix epoch
+                    var unixTime = (long)(timestamp.Subtract(new DateTime(1970, 1, 1))).TotalMilliseconds;
+                    props.Timestamp = new AmqpTimestamp(unixTime);
+                }
+            }
+
+            if (headers.TryGetValue(RabbitMqHeaders.Type, out var type))
+            {
+                props.Type = type;
+            }
+
+            // Alternative way of setting RabbitMqHeaders.DeliveryMode
+            if (!headers.ContainsKey(RabbitMqHeaders.DeliveryMode))
+            {
+                var express = headers.ContainsKey(Headers.Express);
+                props.Persistent = !express;
+            }
+
+            // must be last because the other functions on the headers might change them
+            props.Headers = headers
+                .ToDictionary(kvp => kvp.Key, kvp => (object) HeaderValueEncoding.GetBytes(kvp.Value));
+
+            return props;
         }
 
-        void EnsureQueueIsInitialized(string destinationAddress, IModel model)
+        void EnsureQueueExists(string destinationAddress, IModel model)
         {
-            _initializedQueues
+            _verifiedQueues
                 .GetOrAdd(destinationAddress, _ =>
                 {
                     if (_declareInputQueue)
                     {
-                        DeclareQueue(destinationAddress, model, true);
-                    }
-
-                    if (_bindInputQueue)
-                    {
-                        BindInputQueue(destinationAddress, model);
+                        // Checks if queue exists, throws if the queue doesn't exist
+                        model.QueueDeclarePassive(destinationAddress);
```

Comment:
Does this throw an exception if the queue does not exist? 

We should probably catch & wrap the exception, saying something like "the queue 'dev-null' does not exist"